### PR TITLE
docs: add FAQs for core Firehose related queries: FAQ PR #6

### PR DESCRIPTION
**Pull request by Meghajit**
This PR does the following:-

builds upon the FAQ section as created in the [FAQ PR # 5](https://github.com/odpf/firehose/pull/141)
adds FAQs for core Firehose related queries and references them in the Table of Contents

### CODE REVIEW

File: docs/reference/faq/index.md [hunk 1]
```diff
@@ -131,4 +211,651 @@ for more details.
...
+   mapped to the header name.
+
+### General Queries
+
+#### Which Java versions firehose work with?
+
+Firehose has been developed and tested to work with Java 8. Compatibility with lower or higher versions is not known and
+hence, needs to be tested prior to use.
+
+#### How does the execution work? Is the entire process sync or async ?
```
Comment (lavkesh) [inline docs/reference/faq/index.md:599]: @Meghajit I think this is outdated.. do yo wanna mention async consumer too?

File: docs/reference/faq/index.md [hunk 2]
```diff
@@ -131,4 +211,651 @@ for more details.
...
+Kafka server to your database/service endpoint. Having provided all the configuration parameters of the sink, Firehose
+will create, manage and monitor one for you. It also automatically scales to match the throughput of your data and
+requires no ongoing administration.
+
+#### Can I do any transformations before sending data to sink, for example filtering ?
+
+Yes, Firehose provides JEXL based filters based on the fields in key or message of the Kafka record. Read
+the [Filters](https://github.com/odpf/firehose/blob/main/docs/concepts/filters.md) section for further details.
+
+#### How to optimise parallelism based on input rate of Kafka messages? Does it depend on sink ?
```
Comment (lavkesh) [inline docs/reference/faq/index.md:744]: It might be better to use asysnc consumer. I think we should remove application_thread in later release.

Comment (Meghajit) [reply docs/reference/faq/index.md:747]: @lavkesh Linked to the consumer documentation

File: docs/reference/faq/index.md (hunk 1 above)
Comment (Meghajit) [reply docs/reference/faq/index.md:600]: @lavkesh Added a reference to the AsyncConsumer documentation.

**Pull request merged**